pilot/pkg/model/config.go

```diff
@@ -127,14 +127,24 @@ type ConfigMeta struct {
 	CreationTimestamp time.Time `json:"creationTimestamp,omitempty"`
 }
 
-// Config is a configuration unit consisting of the type of configuration, the
-// key identifier that is unique per type, and the content represented as a
-// protobuf message.
+// Override of the ProtoMessage such that gogo can be rmeoved
+type Message interface {
+	Reset()
+	String() string
+	ProtoMessage()
+}
+
+// Config is a configuration unit consisting of the google.golang.org/protobtufype of configuration, the
+// key identifier that is unique per type, and the content represented as a protobuf message.
 type Config struct {
 	ConfigMeta
+	Spec Message
```

Comment:
I am not actually sure this changes anything? Our `Message` interface is exactly the same as the previous proto.Message?

Reply:
agree - I was just pushing my WIP in the event others wish to follow along as I flog the various repositories.